src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -3183,7 +3189,7 @@ public virtual bool CoercesTo(DType typeDest, out bool isSafe, out DType coercio
                                  Boolean.Accepts(this);
                     break;
                 case DKind.String:
-                    doesCoerce = Kind != DKind.Color && Kind != DKind.Control && Kind != DKind.DataEntity && Kind != DKind.OptionSet && Kind != DKind.View && Kind != DKind.Polymorphic && Kind != DKind.File && Kind != DKind.LargeImage;
+                    doesCoerce = Kind != DKind.Color && Kind != DKind.Control && Kind != DKind.DataEntity && Kind != DKind.OptionSet && Kind != DKind.View && Kind != DKind.Polymorphic && Kind != DKind.File && Kind != DKind.LargeImage && Kind != DKind.Void;
```

Comment:
>Kind != DKind.Void [](http://example.com/codeflow?start=237&length=18)

Similar to Accepts: Void is special enough that it should have its own treatment at the beginning of this method.